pkg/apis/core/validation/shoot.go

```diff
@@ -1443,3 +1463,20 @@ func ValidateContainerRuntimes(containerRuntime []core.ContainerRuntime, fldPath
 
 	return allErrs
 }
+
+// ValidateFeatureGates validates the given Kubernetes feature gates against the given Kubernetes version.
+func ValidateFeatureGates(featureGates map[string]bool, version string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for featureGate := range featureGates {
+		supported, err := kutil.IsFeatureGateSupported(featureGate, version)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, featureGate, err.Error()))
+		}
+		if !supported {
+			allErrs = append(allErrs, field.Invalid(fldPath, featureGate, fmt.Sprintf("not supported for Kubernetes version %s", version)))
```

Comment:
> // Invalid returns a *Error indicating "invalid value".  This is used
// to report malformed values (e.g. failed regex match, too long, out of bounds).

> // Forbidden returns a *Error indicating "forbidden".  This is used to
// report valid (as per formatting rules) values which would be accepted under
// some conditions, but which are not permitted by current conditions (e.g.
// security policy).

So I guess it should be `Invalid` in case of an error (unknown feature gates should also be reported as errors) and `Forbidden` otherwise (a known feature gate that is not supported for the given version).

Reply:
Done as proposed above, let me know if you still have concerns.